Update authors and relnotes for 1.1

brson opened the pull request:
[Randard](https://github.com/brson/rust/blob/relnotes/RELEASES.md).

I need help identifying language things and highlights, and I am kind of tired of the `~X changes, numerous bugfixes` thing, which was cute for a few years.

cc @aturon @nrc @pnkfelix

rust-highfive commented:
r? @nikomatsakis

(rust_highfive has picked a reviewer for you, use r? to override)

bstrie commented:
We should get into the habit of having a "potential breaking changes" section. Even if the section is empty for 1.1, it's good to have the section there anyway so that people know there aren't any.

amaranth commented on RELEASES.md:
```diff
@@ -1,3 +1,117 @@
+  Windows, symlinks can be created with
+  `std::os::windows::symlink_dir` and
+  `std::os::windows::symlink_file`.
+* The `mpsc::Reciever` typo can now be converted into an iterator with
```
Looks like this should be "type" and not "typo".

bstrie replied:
Freudian slip? :) "reciever" -> "receiver", "typo" -> "type"

brson replied:
Funny.

amaranth commented on RELEASES.md:
```diff
@@ -1,3 +1,117 @@
+  [panics on overflow when debug assertions are enabled][abs].
+* The [`Cloned`] iterator, which was accidentally left unstable for
+  1.0 [has been stabilized][c].
+* The [`Incoming`] iterator, which iterators over incoming TCP
```
Should be "iterates" instead of "iterators".

amaranth commented on RELEASES.md:
```diff
@@ -1,3 +1,117 @@
+  properly exported][inc].
+* [`BinaryHeap`] no longer corrupts itself [when functions called by
+  `sift_up` or `sift_down` panic][bh].
+* The [`split_off`] method of `LinkedList` longer corrupts
```
Missing a [ for the link, looks like this should be "[no longer corrupts".

bstrie commented:
Under compiler speed, we should also mention https://github.com/rust-lang/rust/pull/25323 for its speed and memory improvements.

bluss commented:
@bstrie We need an appropriate way to word it. Maybe “changes to type inference”, or “minor breaking changes”

bluss commented on RELEASES.md:
```diff
@@ -1,3 +1,117 @@
+  whitespace boundaries.
+* The [`Iterator::cloned`] method creates an iterator of `T` from
+  an iterator of `&T` by cloning each element. It is equivalent
+  to `.map(|&x| x)`.
```
I think `.cloned()` was already present in Rust 1.0? If not, move this down / up so that it is next to mention of `Cloned` below.

Stebalien replied:
Pedantic nit: That statement isn't quite correct. `.cloned()` equivalent (<=>) to `.map(|x| x.clone())`. It only replaces/supersedes `.map(|&x| x)`.

sfackler replied:
Yep, `.cloned()` was already stable in 1.0.

bluss commented on RELEASES.md:
```diff
@@ -1,3 +1,117 @@
+  to rustc.
+* [Android executables are always position independent][pie].
+* [The `drop_with_repr_extern` lint warns about mixing `extern(C)`
+  with `Drop`][drop].
```
I think this should say `repr(C)`.

alexcrichton commented on AUTHORS.txt:
```diff
@@ -1,7 +1,7 @@
 Rust was written by these fine people:
 
-A.J. Gardner <mrhota@users.noreply.github.com>
```
It looks like there was some churn in this file due to some differences in how things are sorted, not sure if that was intentional or not.

alexcrichton replied:
Ah if this was all generated though then that's fine to just rely on that sorting order from now on.

brson replied:
Yes, sorry. The sort order changes depending on whether the person updating it is  sorting on the shell or in an editor. I added a script to automate this process so hopefully we can stop the churn.

alexcrichton commented on RELEASES.md:
```diff
@@ -1,3 +1,117 @@
+
+* Type checking performance [has improved notably][sk].
+* The compiler [suggests code changes][ch] for more errors.
+* Support for LLVM 3.5 and earlier has been removed.
```
I believe we actually still have support for 3.5?

brson replied:
Looks like it. I swear I saw a patch.

nrc commented:
I have made exactly zero language changes this cycle, sorry.

nrc commented on RELEASES.md:
```diff
@@ -1,3 +1,117 @@
+
+* TODO: There have been changes to dropck that should probably be
+  mentioned
+
```
Are DST coercions (#24619) in this release?

bluss replied:
Judging from the API docs I think so -- all the `T: ?Sized` changes to `Rc<T>` are in.

bors commented:
:umbrella: The latest upstream changes (presumably #25844) made this pull request unmergeable. Please resolve the merge conflicts.

brson commented:
Rebased and addressed all comments, except that I did not add a section on breaking changes as suggested by @bstrie, though I do think we should list all breaking changes every release. We need a better way to track them. I *think* there were minor breaking changes this release that were considered acceptable and to affect no code, but I don't recall them offhand.

brson commented:
r? @aturon

aturon commented on RELEASES.md:
```diff
@@ -13,8 +13,103 @@ Libraries
+  properly exported][inc].
+* [`BinaryHeap`] no longer corrupts itself [when functions called by
+  `sift_up` or `sift_down` panic][bh].
+* The [`split_off`] method of `LinkedList` [longer corrupts
```
no longer?

aturon commented on RELEASES.md:
```diff
@@ -13,8 +13,103 @@ Libraries
+----
 
-[fs-expand]: https://github.com/rust-lang/rust/pull/25844
+* Type checking performance [has improved notably][sk] with
```
We may want to call out perf improvements in the highlights section.

brson replied:
Done.

brson commented:
@bors r=aturon

bors commented:
:pushpin: Commit f1a029a has been approved by `aturon`

<!-- @bors r=aturon f1a029a2bc4fccf62a3306ccf6495e2f80a56eb9 -->

brson commented:
@bors r=aturon

bors commented:
:pushpin: Commit ab0bb09 has been approved by `aturon`

<!-- @bors r=aturon ab0bb0956d1d3cbf67ace0c9c215a1730400b2de -->

bors commented:
:hourglass: Testing commit ab0bb09 with merge a951569...

bors commented:
:sunny: Test successful - [auto-linux-32-nopt-t](http://buildbot.rust-lang.org/builders/auto-linux-32-nopt-t/builds/5421), [auto-linux-32-opt](http://buildbot.rust-lang.org/builders/auto-linux-32-opt/builds/5432), [auto-linux-64-nopt-t](http://buildbot.rust-lang.org/builders/auto-linux-64-nopt-t/builds/5383), [auto-linux-64-opt](http://buildbot.rust-lang.org/builders/auto-linux-64-opt/builds/5410), [auto-linux-64-x-android-t](http://buildbot.rust-lang.org/builders/auto-linux-64-x-android-t/builds/5379), [auto-mac-32-opt](http://buildbot.rust-lang.org/builders/auto-mac-32-opt/builds/5453), [auto-mac-64-nopt-t](http://buildbot.rust-lang.org/builders/auto-mac-64-nopt-t/builds/5446), [auto-mac-64-opt](http://buildbot.rust-lang.org/builders/auto-mac-64-opt/builds/5414), [auto-win-gnu-32-nopt-t](http://buildbot.rust-lang.org/builders/auto-win-gnu-32-nopt-t/builds/397), [auto-win-gnu-32-opt](http://buildbot.rust-lang.org/builders/auto-win-gnu-32-opt/builds/397), [auto-win-gnu-64-nopt-t](http://buildbot.rust-lang.org/builders/auto-win-gnu-64-nopt-t/builds/396), [auto-win-gnu-64-opt](http://buildbot.rust-lang.org/builders/auto-win-gnu-64-opt/builds/396)
